MoreLinq.Test/ScanRightTest.cs

```diff
@@ -42,11 +42,14 @@ public void ScanRightFuncIsNotInvokedOnSingleElementSequence()
             Assert.That(result, Is.EqualTo(new[] { value }));
         }
 
-        [Test]
-        public void ScanRight()
+        [TestCase(SourceKind.List)]
```

Comment:
I don't think there is a way to prove that this test case can be broken? `ScanRight` internally skips materialisation to a list if the source is already one. Ideally, if that optimisation is removed then the test should break but detecting whether a copy was made or not is brittle. I'd hate to rely on current internal implementation details like it eventually ends up calling [`ICollection<>`'s `CopyTo`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.icollection-1.copyto?), which [our `BreakingCollection<>` implements](https://github.com/morelinq/MoreLINQ/blob/bab96ace806efba59c444f8bf31faa36b6aefe57/MoreLinq.Test/BreakingCollection.cs#L45) (and why the test doesn't break).

Reply:
Something really hacky we could do is add a `ThrowOnCopy` property to `Breaking(ReadOnly)Collection` (which would toggle `CopyTo`'s behavior), and then do something like this:

```
var enumerable = Enumerable.Range(1, 5).Select(x => x.ToString()).ToSourceKind(sourceKind);
switch (enumerable)
{
case BreakingReadOnlyCollection readOnlyCollection: 
    readOnlyCollection.ThrowOnCopy = true; break;
case BreakingCollection collection:
    collection.ThrowOnCopy = true; break;
}
```

Reply:
As I said, I'd hate to rely on current internal implementation details that can change (and [already have I believe](https://github.com/dotnet/corefx/pull/3429)) so I think anything we'll come up with will seem like a hack at best. I'm not sure it's worth the trouble. The best option might be to comment out that test case and make a note of why.